src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -257,58 +254,53 @@ internal static ProcessInfo[] GetProcessInfos(Predicate<int>? processIdFilter =
             ProcessInfo[] processInfos;
 
             // Start with the default buffer size.
-            int bufferSize = DefaultCachedBufferSize;
+            int bufferSize = DefaultCachedBufferSize * sizeof(long);
 
             // Get the cached buffer.
-            long[]? buffer = Interlocked.Exchange(ref CachedBuffer, null);
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(bufferSize);
```

Comment:
Shouldn't the rented buffer size account for potential alignment adjustments?

Reply:
When we use the default size it's just a guess, and even 63 bytes less (`8 * 128 * 1024 - 63`) should be enough.

When it comes to the case when we fail and the OS tells us how much we should allocate, we already add an extra 10kb to it just in case:

https://github.com/dotnet/runtime/blob/fb5ea4bbacd83972cbcf704d1a0622d50cab5d48/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs#L323-L325

and here again even `-63` bytes should be enough

Reply:
Sounds reasonable, thanks.